libp2p/host/ping.py

```diff
@@ -45,16 +66,65 @@
     return True
 
 
-async def handle_ping(stream: INetStream) -> None:
-    """``handle_ping`` responds to incoming ping requests until one side errors
-    or closes the ``stream``."""
-    peer_id = stream.muxed_conn.peer_id
+class PingService:
+    """PingService executes pings and returns RTT in miliseconds."""
 
-    while True:
+    def __init__(self, host: IHost):
+        self._host = host
+
+    async def ping(self, peer_id: PeerID) -> int:
+        stream = await self._host.new_stream(peer_id, (ID,))
         try:
-            should_continue = await _handle_ping(stream, peer_id)
-            if not should_continue:
-                return
+            rtt = await _ping(stream)
+            await _close_stream(stream)
+            return rtt
         except Exception:
-            await stream.reset()
-            return
+            await _close_stream(stream)
+            raise
+
+    async def ping_loop(
+        self, peer_id: PeerID, ping_amount: Union[int, float] = math.inf
+    ) -> "PingIterator":
+        stream = await self._host.new_stream(peer_id, (ID,))
```

Comment:
would be helpful to leave a docstring describing why we have `ping` and `ping_loop`.

it turns out that `ping` is just `ping_loop` specialized to `ping_amount == 1`. which also suggest we may be able to simplify the implementation here...

Reply:
Okay, it is a good idea to write a docstring!

The reason why I thought of separating into two methods is because `ping_loop` is a way to use it as an intuitive `for loop`. However, it is not so straightforward, you must generate the iterator with `ping_loop` and then construct a `for loop`. So I created the second method `ping` for when you just want a quick, simple and one-line ping (maybe for a rapid keep-alive, or something else). 

If am open to discussion whether it really is worth it or useful or not :)

Reply:
Also the loop is useful in case you want to add some more functionality between each ping. (or maybe it is not a good feature?)

However, I have been thinking and maybe it is a good idea to add a second argument to the `ping(peer_id)`, which specifies the amount of pings and return a list of `RTTs`. Something like `rtts = ping_service.ping(peer_id, amount=5)`.